### PR TITLE
update to support 1.20.4

### LINKED ISSUE #159: Update to 1.20.2
Pls update to 1.20.2. The [fabric-carpet](https://github.com/gnembon/fabric-carpet/releases/tag/1.4.118) release has already been released

### PR DESCRIPTION
**Pull request by petersv5**
This is an early stab at updating to 1.20.4. There are several deficiencies:

- carpetfixes.mixins.advanced.MinecraftServer_autosaveDelayMixin needs to be rewritten to match the changes in the game

- I do not understand what carpetfixes.mixins.reIntroduced.PlayerManager_LlamaRidingDupeMixin is trying to do with the "vmp" and "c2me" prefixes.

- The method call intercept to com.google.commons.collect.Maps.newHashMap() fail in carpetfixes.mixins.blockFixes.PistonBlock_pushOrderMixin and I do not understand why.

- Update suppression fixes should be removed as they are handled by TIS.

The mod does compile and work for the few fixes I have tried.

This should eventually resolve #159.

### CODE REVIEW

**Comment by FxMorin** [comment]
Will be taking a look at this today

File: src/main/java/carpetfixes/mixins/advanced/MinecraftServer_autosaveDelayMixin.java
Comment (FxMorin) [inline src/main/java/carpetfixes/mixins/advanced/MinecraftServer_autosaveDelayMixin.java:1]: Ill be removing it, as minecraft has done significant work on it to fix lag issues.

**Review by FxMorin — APPROVED** [review]
I'll probably merge this and make the rest of the changes on the branch directly

**Comment by FxMorin** [comment]
> carpetfixes.mixins.advanced.MinecraftServer_autosaveDelayMixin needs to be rewritten to match the changes in the game

I'll be removing it.


> I do not understand what carpetfixes.mixins.reIntroduced.PlayerManager_LlamaRidingDupeMixin is trying to do with the "vmp" and "c2me" prefixes.

I'll also be removing that rule. Those prefixes are for the vmp (very many people) & c2me (Concurrent Chunk Management Engine) mods. It's a compatibility fix.


> The method call intercept to com.google.commons.collect.Maps.newHashMap() fail in carpetfixes.mixins.blockFixes.PistonBlock_pushOrderMixin and I do not understand why.

Ya mixin always does this for google commons, not sure why. However it will still inject correctly.

**Comment by petersv5** [comment]
> > The method call intercept to com.google.commons.collect.Maps.newHashMap() fail in carpetfixes.mixins.blockFixes.PistonBlock_pushOrderMixin and I do not understand why.
> 
> Ya mixin always does this for google commons, not sure why. However it will still inject correctly.

I thought I was going crazy. :-)

**Pull request merged**